### ISSUE TITLE
Reproducing vanishing _kombu.binding.reply.celery.pidbox and RuntimeError: pubsub connection not set

### ISSUE THREAD

**Issue by krisztian-fl** [celery/kombu#1365]
In regard with https://github.com/celery/kombu/issues/1362 and other similar issues, I've managed to reproduce the problem with the following method. I noticed that newly forked workers beyond the initial N workers are having trouble reliably using the control interface, and during testing that came this simple approach that fails for me every single time in about 30-90 seconds, reliably producing `OperationalError: Cannot route message for exchange 'reply.celery.pidbox': Table empty or key no longer exists` followed soon by a `RuntimeError('pubsub connection not set: did you forget to call subscribe() or psubscribe()?')`.

Create a simple task that is using the control interface:
```
@app.task(bind=True)
def fork_task(self):
    i = self.app.control.inspect()
    a = i.active()
    if a is None:
        print(f"A:'NOTFOUND'", end='')
        fork_task.apply_async()
        return
    print(f"A:'FOUND'", end='')
    fork_task.apply_async(countdown=2)
```

Run a worker that guarantees frequent worker spawning, then queue up the task a few dozen or so times:
`celery -A app worker -c 4 --max-tasks-per-child 8`
`[fork_task.delay() for _ in range(24)]`

Memory available to Redis and its eviction policy does not seem to affect the outcome, I've tried it with 32-128MB and `allkeys-lru` and `allkeys-lfu`.

I hope it helps narrowing down the problem.

Redis == 6.0.8
Celery == 5.1.2
Kombu == 5.1.0
Billiard == 3.6.4.0

**Comment by matusvalo** [comment]
#1362 was closed in favor of #1063.

**Comment by matusvalo** [comment]
Issue #1377 seems to be related with this issue. It was fixed by #1378. Could someone double check whether this issue is also fixed by this PR? Please let us know here in comments.

**Comment by matusvalo** [comment]
This bug seems to be fixed by #1378. For now I am closing it.
